NuRadioReco/framework/base_trace.py

```diff
@@ -69,32 +70,65 @@ def get_frequency_spectrum(self):
             self.__time_domain_up_to_date = False
         return np.copy(self._frequency_spectrum)
 
-    def set_trace(self, trace, sampling_rate):
+    def set_trace(self, trace, sampling_rate=None):
```

Comment:
Why have a default argument for sampling_rate if that raises an error?

Reply:
see https://rno-g.slack.com/archives/C048SKA1V8V/p1705670362971339

Reply:
Sorry, I still don't follow. You need to pass either a float or the string "same". So this is not an optional argument, why give it a default value?

Reply:
fine